lib/std/os/windows/user32.zig

```diff
@@ -71,105 +113,561 @@ pub const WM_MBUTTONDOWN = 0x0207;
+pub const WS_CHILDWINDOW = WS_CHILD;
+
+pub const WS_EX_DLGMODALFRAME = 0x00000001;
+pub const WS_EX_NOPARENTNOTIFY = 0x00000004;
+pub const WS_EX_TOPMOST = 0x00000008;
+pub const WS_EX_ACCEPTFILES = 0x00000010;
+pub const WS_EX_TRANSPARENT = 0x00000020;
+pub const WS_EX_MDICHILD = 0x00000040;
+pub const WS_EX_TOOLWINDOW = 0x00000080;
+pub const WS_EX_WINDOWEDGE = 0x00000100;
+pub const WS_EX_CLIENTEDGE = 0x00000200;
+pub const WS_EX_CONTEXTHELP = 0x00000400;
+pub const WS_EX_RIGHT = 0x00001000;
+pub const WS_EX_LEFT = 0x00000000;
+pub const WS_EX_RTLREADING = 0x00002000;
+pub const WS_EX_LTRREADING = 0x00000000;
+pub const WS_EX_LEFTSCROLLBAR = 0x00004000;
+pub const WS_EX_RIGHTSCROLLBAR = 0x00000000;
+pub const WS_EX_CONTROLPARENT = 0x00010000;
+pub const WS_EX_STATICEDGE = 0x00020000;
+pub const WS_EX_APPWINDOW = 0x00040000;
+pub const WS_EX_LAYERED = 0x00080000;
+pub const WS_EX_OVERLAPPEDWINDOW = (WS_EX_WINDOWEDGE | WS_EX_CLIENTEDGE);
+pub const WS_EX_PALETTEWINDOW = (WS_EX_WINDOWEDGE | WS_EX_TOOLWINDOW | WS_EX_TOPMOST);
+
+pub const CW_USEDEFAULT = @bitCast(i32, @as(u32, 0x80000000));
+
+pub extern "user32" fn CreateWindowExA(dwExStyle: DWORD, lpClassName: [*:0]const u8, lpWindowName: [*:0]const u8, dwStyle: DWORD, X: i32, Y: i32, nWidth: i32, nHeight: i32, hWindParent: ?HWND, hMenu: ?HMENU, hInstance: HINSTANCE, lpParam: ?LPVOID) callconv(WINAPI) ?HWND;
+pub fn createWindowExA(dwExStyle: u32, lpClassName: [*:0]const u8, lpWindowName: [*:0]const u8, dwStyle: u32, X: i32, Y: i32, nWidth: i32, nHeight: i32, hWindParent: ?HWND, hMenu: ?HMENU, hInstance: HINSTANCE, lpParam: ?*c_void) !HWND {
+    const window = CreateWindowExA(dwExStyle, lpClassName, lpWindowName, dwStyle, X, Y, nWidth, nHeight, hWindParent, hMenu, hInstance, lpParam);
+    if (window) |win| return win;
+
+    return switch (GetLastError()) {
+        .CLASS_DOES_NOT_EXIST => error.ClassDoesNotExist,
+        .INVALID_PARAMETER => unreachable,
+        else => |err| return windows.unexpectedError(err),
+    };
+}
+
+pub extern "user32" fn CreateWindowExW(dwExStyle: DWORD, lpClassName: [*:0]const u16, lpWindowName: [*:0]const u16, dwStyle: DWORD, X: i32, Y: i32, nWidth: i32, nHeight: i32, hWindParent: ?HWND, hMenu: ?HMENU, hInstance: HINSTANCE, lpParam: ?LPVOID) callconv(WINAPI) ?HWND;
+pub var pfnCreateWindowExW: @TypeOf(RegisterClassExW) = undefined;
+pub fn createWindowExW(dwExStyle: u32, lpClassName: [*:0]const u16, lpWindowName: [*:0]const u16, dwStyle: u32, X: i32, Y: i32, nWidth: i32, nHeight: i32, hWindParent: ?HWND, hMenu: ?HMENU, hInstance: HINSTANCE, lpParam: ?*c_void) !HWND {
+    const function = selectSymbol(CreateWindowExW, pfnCreateWindowExW, .win2k);
+    const window = function(dwExStyle, lpClassName, lpWindowName, dwStyle, X, Y, nWidth, nHeight, hWindParent, hMenu, hInstance, lpParam);
+    if (window) |win| return win;
+
+    return switch (GetLastError()) {
+        .CLASS_DOES_NOT_EXIST => error.ClassDoesNotExist,
```

Comment:
IIRC you should prefer to put the `return` on the line where the error is created, as it makes the error tracebacks better?